metron-platform/metron-writer/src/main/java/org/apache/metron/writer/FlushPolicy.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.metron.writer;
+
+import org.apache.metron.common.configuration.writer.WriterConfiguration;
+import org.apache.metron.common.writer.BulkMessageWriter;
+import org.apache.metron.common.writer.BulkWriterMessage;
+import org.apache.metron.common.writer.BulkWriterResponse;
+
+import java.util.List;
+
+/**
+ * This interface is used by the {@link org.apache.metron.writer.BulkWriterComponent} to determine if a batch should be flushed
+ * and handle the {@link org.apache.metron.common.writer.BulkWriterResponse} when a batch is flushed.
+ */
+public interface FlushPolicy {
+
+  /**
+   * This method is called whenever messages are passed to {@link BulkWriterComponent#write(String, BulkWriterMessage, BulkMessageWriter, WriterConfiguration)}.
+   * Each implementation of {@link org.apache.metron.writer.FlushPolicy#shouldFlush(String, WriterConfiguration, int)} will be called in order
+   * and the first one to return true will trigger a flush and continue on.
+   * @param sensorType sensor type
+   * @param configurations configurations
+   * @param batchSize batch size of messages written
+   * @return true if batch should be flushed
+   */
+  boolean shouldFlush(String sensorType, WriterConfiguration configurations, int batchSize);
```

Comment:
Why pass in the `batchSize`, rather than just pass in the batch itself?  It seems like passing the entire 'batch' of messages would give policies more information, at little to no extra cost.
```
boolean shouldFlush(
  String sensorType, 
  WriterConfiguration configs, 
  List<BulkWriterMessage<MESSAGE_T>> messages)
```

Reply:
I did it that way initially but changed it based on this [feedback](https://github.com/apache/metron/pull/1330#pullrequestreview-203467469) from @ottobackwards.  I see both points so I could go either way.

Reply:
I think it just makes sense to provide the information in as raw a form as possible to the policies, rather than assuming that they need certain bits of information, like the batch size. 

I don't understand the downside to that approach in @ottobackwards 's original comment.

Reply:
The latest commit changes it back to the way I had it initially.  The `FlushPolicy.shouldFlush` method now accepts a list of messages.